tests/venice-pulsar-test/src/pulsarintegrationtest/java/sink/PulsarVeniceSinkTest.java

```diff
@@ -0,0 +1,281 @@
+      LOGGER.info("Stopping docker environment");
+      environment.stop();
+      LOGGER.info("Docker environment stopped");
+    }
+  }
+
+  @Test
+  public void testPulsarVeniceSink() throws Exception {
+    LOGGER.info("testPulsarVeniceSink");
+
+    Schema<Person> personSchema = Schema.AVRO(Person.class);
+    String schema = personSchema.getSchemaInfo().getSchemaDefinition();
+
+    String keyAsvc = "{\"name\": \"key\",\"type\": \"string\"}";
+    String valueAsvc = schema;
+
+    String veniceControllerUrl = "http://venice-controller:5555";
+    String veniceRouterUrl = "http://venice-router:7777";
+    String keyFile = "/tmp/key.asvc";
+    String valueFile = "/tmp/value.asvc";
+    String jar = "/opt/venice/bin/venice-admin-tool-all.jar";
+    String clusterName = "venice-cluster0";
+    String storeName = "t1_n1_s1";
+
+    LOGGER.info("Setting up Venice");
+    saveKeyValueSchemaFiles(keyAsvc, valueAsvc, keyFile, valueFile);
+    createVeniceStore(veniceControllerUrl, keyFile, valueFile, jar, clusterName, storeName);
+    updateVeniceStoreQuotas(veniceControllerUrl, jar, clusterName, storeName);
+    initVeniceStore(veniceControllerUrl, jar, clusterName, storeName);
+
+    LOGGER.info("Setting up Pulsar");
+    createNamespace();
+    setRetention();
+    createTopics();
+
+    LOGGER.info("Starting up Pulsar Venice Sink");
+    String token =
+        "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJzdXBlcnVzZXIifQ.JQpDWJ9oHD743ZyuIw55Qp0bb8xzP6gK0KIWRniF2WnJB1m3v5MsrpfMlmRIlFc3-htWRAFHCc4E0ipj7JU8HjBqLIvVErRseRG-UTM1EprVkj0mk37jXV3ef7gER0KHn9CUKEQPfmTACeKlQ2oV4_qPAZ6HiEt51vzANfZH24vLCIjiOG77Z4s_w2sfgpiodRmhBLFOg_qnQTfGs7TBDWgu4DRoJ6CYZSEcp8q7j8xp_zNVIFGTRjWskocUvedHS9ZsCGZjzuPvRPp19B0VvAjEjtwpa6j7Khvjf4imjp2QHDnZwpCIEp4DSicwM48F5q4k722IdiyTTsVBWy8Cyg";
+
+    String sinkConfig = "{\"veniceDiscoveryUrl\":\"" + veniceControllerUrl + "\"," + "\"storeName\":\"t1_n1_s1\","
+        + "\"kafkaSaslMechanism\":\"PLAIN\",\"kafkaSecurityProtocol\":\"SASL_PLAINTEXT\","
+        + "\"kafkaSaslConfig\":\"org.apache.kafka.common.security.plain.PlainLoginModule "
+        + "required username=\\\"public\\\" password=\\\"token: " + token + "\\\"\\\";\\\");\"}";
+
+    ExecResult createSinkRes = execByServiceAsssertNoStdErr(
+        "proxy",
+        "bash",
+        "-c",
+        "/pulsar/bin/pulsar-admin sinks create --tenant t1 --namespace n1 "
+            + "--name venice -t venice -i t1/n1/input --sink-config '" + sinkConfig + "'");
+    assertTrue(createSinkRes.getStdout().contains("Created successfully"));
+
+    // Awaitility.await().atMost(90, TimeUnit.SECONDS).untilAsserted(() -> {
+    // ExecResult res = execByService(
+    // "venice-client",
+    // "bash",
+    // "-c",
+    // "java -jar " + jar + " --describe-store --url " + veniceControllerUrl
+    // + " --cluster " + clusterName
+    // + " --store " + storeName);
+    //
+    // assertTrue(res.getStdout().contains("\"status\" : \"ONLINE\""));
+    // });
```

Comment:
Could you try writing an assertion that verifies the `storeName_rt` topic is created? The expected behavior is that it would be created already by the time we get here, but if that's not the case, then it gives us a good clue on what is going wrong in this scenario...

If that is indeed the problem, then you could wrap the assertion into an `await()` call. In theory, that should be the minimum wait time required for the test to succeed (i.e. without the subsequent sleep). Even if it does pass with the await, that is just a bandaid, and we should still fix the root cause of why the RT wasn't already created by the time we got here.

Reply:
I tried
```java
     Awaitility.await().atMost(30, TimeUnit.SECONDS).untilAsserted(() -> {
       ExecResult res = execByService(
             "kafka",
             "bash",
             "-c",
             "/opt/kafka/bin/kafka-topics.sh --list --bootstrap-server localhost:9092");

       assertTrue(res.getStdout().contains(storeName + "_rt"));
     });
```

This assertion passes imemdiatley
```
INFO [PulsarVeniceSinkTest] [awaitility-thread]         kafka Exec
        [bash, -c, /opt/kafka/bin/kafka-topics.sh --list --bootstrap-server localhost:9092]
        StdOut:
    t1_n1_s1_rt
    t1_n1_s1_v1

        StdErr:

```

The test failed, get didn't find the entry in Venice (`value=null`)
```
INFO [PulsarVeniceSinkTest] [awaitility-thread]         venice-client Exec
        [bash, -c, java -jar /opt/venice/bin/venice-thin-client-all.jar t1_n1_s1 name0 http://venice-router:7777 false ""]
        StdOut:

    com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_9/MultiGetResponseRecordV1_GenericDeserializer_1339531883776153861_1339531883776153861.java
    Key string parsed successfully. About to make the query.
    key-class=java.lang.String
    value-class=null
    request-path=storage/t1_n1_s1/Cm5hbWUw?f=b64
    key=name0
    value=null
```

Reply:
Ok thanks, so that's not the issue then. Will keep digging.